compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -589,12 +589,20 @@ convType env reexported =
             args <- mapM (convType env reexported) lfArgs
             pure $ HsParTy noExt (noLoc $ foldl (HsAppTy noExt . noLoc) tyvar (map noLoc args))
 
+        LF.TApp (LF.TCon LF.Qualified {..}) (LF.TStruct fields)
+            | qualModule == LF.ModuleName ["DA", "Internal", "PromotedText"]
+            , ["PromotedText"] <- LF.unTypeConName qualObject
+            , [(LF.FieldName text, LF.TUnit)] <- fields
```

Comment:
Should we error out if it is not a struct of the form that we expect instead of falling into the next case?

Reply:
Done!